k8s/k8s.go

```diff
@@ -0,0 +1,94 @@
+package k8s
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/libopenstorage/secrets"
+)
+
+const (
+	Name          = "k8s"
+	K8sSecretPath = "/etc/secrets/"
+)
+
+type k8sSecrets struct{}
+
+func getSecretKey(secretId string) string {
+	return K8sSecretPath + secretId
+}
+
+func New(
+	secretConfig map[string]interface{},
+) (secrets.Secrets, error) {
+	return &k8sSecrets{}, nil
+}
+
+func (s *k8sSecrets) String() string {
+	return Name
+}
+
+func (s *k8sSecrets) GetSecret(
+	secretId string,
+	keyContext map[string]string,
+) (map[string]interface{}, error) {
+	cipherBlob := []byte{}
+	secretPath := getSecretKey(secretId)
+	_, err := os.Stat(secretPath)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, secrets.ErrInvalidSecretId
+		}
+		return nil, err
+	}
+	cipherBlob, err = ioutil.ReadFile(secretPath)
+	if err != nil || len(cipherBlob) == 0 {
+		return nil, fmt.Errorf("Invalid secretId. Unable to read cipherBlob"+
+			" associated with secretId: %v", err)
+	}
+
+	secretData := make(map[string]interface{})
+	secretData[secretId] = cipherBlob
```

Comment:
Will this work in case of kubernetes? How does the secret data from kubernetes look like in the mounted file?

Reply:
It is in plain text or rather it is in whatever format the user keeps it
I assume it would be the same docker secrets. If the user gives it in base64 then we will return the base64 encoded version in GetSecret.